Server survey 20220929

Cyanoxygen opened the pull request:
Topic Description
-----------------

This topic updates MariaDB, nginx and httpd to their latest version, and makes some modifications on various things, e.g. default configurations, etc.

Related: Issue #4043.

Package(s) Affected
-------------------

- `nginx`
- `httpd`
- `mariadb`

Security Update?
----------------

No

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`

**Secondary Architectures**

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`
- [ ] RISC-V 64-bit `riscv64`

Cyanoxygen committed:
nginx: various configuration rearrangement
- Separate virtual hosts
- use http user
- See issue #4043.

Cyanoxygen committed:
httpd: various configuration rearrangement
- Separate virtual hosts
- use http user
- See issue #4043.

Cyanoxygen committed:
mariadb: add conf.d directory to allow customization

Cyanoxygen committed:
mariadb: migrate datadir to /var/db/mysql
- Add sample config file pointing to correct datadir and include other config files correctly.

Cyanoxygen committed:
nginx: create /srv/www folder during beyond stage

Cyanoxygen committed:
nginx: add sites-available/default to conffiles

Cyanoxygen committed:
nginx: update to 1.23.1

Cyanoxygen committed:
httpd: remove redundant directory creation

remove installed index.html as index.apache.html is the default

Cyanoxygen committed:
nginx: correct the wrong permissions in postinst

Cyanoxygen committed:
httpd: bump REL

Cyanoxygen committed:
httpd: replace apache user with httpd user

also corrected a typo in httpd.conf

Cyanoxygen committed:
httpd: correct the wrong permissions in postinst

Cyanoxygen committed:
httpd: detect existance of /srv/http before prompting

Cyanoxygen committed:
mariadb: update to 10.9.3

Cyanoxygen committed:
mariadb: remove patches (outdated)

Cyanoxygen committed:
mariadb: add prompt after first-time initialization

- And fix mixed indentation

MingcongBai commented on extra-database/mariadb/autobuild/defines:
```diff
@@ -14,7 +14,7 @@ CMAKE_AFTER="-DCMAKE_AR=/usr/bin/gcc-ar \
              -DCMAKE_INSTALL_PREFIX=/usr \
              -DSYSCONFDIR=/etc/mysql \
-             -DMYSQL_DATADIR=/var/lib/mysql \
+             -DMYSQL_DATADIR=/var/db/mysql \
```
We might consider turning this into a QA standard.

Cyanoxygen replied:
All of our RDBMS' should follow this change then.

MingcongBai replied:
> All of our RDBMS' should follow this change then.

Please implement in this PR.

MingcongBai commented on extra-database/mariadb/autobuild/overrides/etc/mysql/my.cnf:
```diff
@@ -1,18 +1,37 @@
+port = 3306
+socket = /run/mysqld/mysqld.sock
 
+# Server only configuration
```
Server configuration *

MingcongBai commented on extra-database/mariadb/autobuild/overrides/etc/mysql/my.cnf:
```diff
@@ -1,18 +1,37 @@
+
+[mysqldump]
+quick
+max_allowed_packet = 50M
```
Why is this set to 50M? Not that I know it's right/wrong, but you should explain what each option does. Including the character set and timezone options, etc.

MingcongBai commented on extra-database/mariadb/autobuild/overrides/etc/mysql/my.cnf:
```diff
@@ -1,18 +1,37 @@
+interactive-timeout
+
+[mysqld_safe]
+open-files-limit = 8192
```
Ditto my notes above.

MingcongBai commented on extra-database/mariadb/autobuild/postinst:
```diff
@@ -1,9 +1,13 @@
+if [[ -e /var/lib/mysql ]]; then
+	# Warn users about the change
+	echo "\`/var/lib/mysql' is no longer used as the default datadir."
+	echo "Your config file stays unmodified."
```
This could be confusing - configuration files are handled by dpkg's conffiles mechanism.

MingcongBai commented on extra-database/mariadb/autobuild/postinst:
```diff
@@ -1,9 +1,13 @@
+	/usr/bin/mysql_install_db --user=mysql --basedir=/usr --datadir=/var/db/mysql
+	echo "========"
+	echo "We have initialized a database at /var/db/mysql."
+	echo "You may run \`systemctl start mysqld.service' to start MariaDB."
```
To launch the MariaDB daemon *

MingcongBai commented on extra-database/mariadb/autobuild/postinst:
```diff
@@ -1,9 +1,13 @@
+	install -dm700 /var/db/mysql
+	chown -R mysql:mysql /var/db/mysql &>/dev/null
+	/usr/bin/mysql_install_db --user=mysql --basedir=/usr --datadir=/var/db/mysql
+	echo "========"
```
Better to use an empty line.

MingcongBai commented on extra-database/mariadb/autobuild/postinst:
```diff
@@ -1,9 +1,13 @@
+	echo "Your config file stays unmodified."
+	echo "Please use \`/var/db/mysql' instead."
+elif [[ ! -e /var/db/mysql ]]; then
+	install -dm700 /var/db/mysql
```
Don't do this. Doing so will result in this directory getting lost in `aoscbootstrap`. Use `tmpfiles.d` instead.

MingcongBai commented on extra-database/mariadb/autobuild/usergroup:
```diff
@@ -1,2 +1,2 @@
 group mysql 89
-user mysql 89 mysql /var/lib/mysql "MariaDB Daemon User" /bin/false
+user mysql 89 mysql /var/db/mysql "MariaDB Daemon User" /bin/false
```
Rewrite this into `sysusers.d`, and add `tmpfiles.d`.

MingcongBai commented on extra-web/httpd/autobuild/beyond:
```diff
@@ -7,12 +7,12 @@ mv -v "$PKGDIR"/usr/bin/suexec \
 abinfo "Creating /srv/www ..."
 mkdir -pv "$PKGDIR"/srv/www
-chown -v -R 25:25 "$PKGDIR"/srv/www
+chown -v -R 207:207 "$PKGDIR"/srv/www
```
Okay, any files you have chown/chgrp'd here should really be handled by `tmpfiles.d`.

MingcongBai commented on extra-web/httpd/autobuild/beyond:
```diff
@@ -40,3 +40,6 @@ abinfo "Installing aosc.layout to /usr/share/doc/httpd ..."
 cp -v "$SRCDIR"/autobuild/aosc.layout \
     "$PKGDIR"/usr/share/doc/httpd/
+
+# index.apache.html is the default index page
```
Use abinfo, and explain better please - what did you do to what effect.

MingcongBai commented on extra-web/httpd/autobuild/overrides/etc/httpd/conf/httpd.conf:
```diff
@@ -0,0 +1,383 @@
+#
```
Where did this file come from and what have you changed?

MingcongBai commented on extra-web/httpd/autobuild/overrides/etc/httpd/conf/modules.conf:
```diff
@@ -0,0 +1,143 @@
+#
```
Ditto notes from last file.

MingcongBai commented on extra-web/httpd/autobuild/overrides/etc/httpd/sites-available/default.conf:
```diff
@@ -0,0 +1,14 @@
+# Default site for Apache
```
Ditto.

MingcongBai commented on extra-web/httpd/autobuild/postinst:
```diff
@@ -1,2 +1,29 @@
+
+chown http:http /srv/www
+chown http:http /srv/www/index.apache.html
+chown -R http:http /srv/www/cgi-bin
```
All of this should be handled by `sysusers.d` and `tmpfiles.d`.

MingcongBai commented on extra-web/httpd/autobuild/postinst:
```diff
@@ -1,2 +1,29 @@
+# Enable default site
+# For Apache httpd the configuration files must be ended with .conf.
+if [ ! -e /etc/httpd/sites-enabled ]; then
+	mkdir -p /etc/httpd/sites-enabled
```
Why not build it into the package? Again, operations like this won't be correctly applied when generating a system release using `aoscboostrap`, as it strips out out-of-package files (barring exceptions).

MingcongBai commented on extra-web/httpd/autobuild/postinst:
```diff
@@ -1,2 +1,29 @@
+fi
+
+if [ ! -e /etc/httpd/sites-enabled/default.conf ]; then
+        ln -sv /etc/httpd/sites-available/default.conf /etc/httpd/sites-enabled/default.conf
```
Ditto above.

MingcongBai commented on extra-web/nginx/autobuild/beyond:
```diff
@@ -5,10 +5,10 @@ install -Dm644 contrib/vim/syntax/nginx.vim \
-    -e '44s|html|/usr/share/nginx/html|' \
-    -e '54s|html|/usr/share/nginx/html|' \
-    -i "$PKGDIR"/etc/nginx/nginx.conf
+# sed -e 's|\<user\s\+\w\+;|user html;|g' \
```
Just drop the lines?

MingcongBai commented on extra-web/nginx/autobuild/beyond:
```diff
@@ -5,10 +5,10 @@ install -Dm644 contrib/vim/syntax/nginx.vim \
+#     -e '54s|html|/usr/share/nginx/html|' \
+#     -i "$PKGDIR"/etc/nginx/nginx.conf
 
 rm "$PKGDIR"/etc/nginx/*.default
```
Lint this script please - missing verbosity and abinfo.

MingcongBai commented on extra-web/nginx/autobuild/conffiles:
```diff
@@ -6,5 +6,6 @@
 /etc/nginx/uwsgi_params
+/etc/nginx/sites-available/default
 /etc/nginx/win-utf
 /etc/logrotate.d/nginx
```
One option would be to just let Autobuild3 handle conffiles generation - drop this file.

MingcongBai commented on extra-web/nginx/autobuild/overrides/etc/nginx/nginx.conf:
```diff
@@ -0,0 +1,57 @@
+user http;
```
Needs more comment explaining what each thing does. Did you get this file from anywhere? If not, where did you learn to write this file?

MingcongBai commented on extra-web/nginx/autobuild/overrides/etc/nginx/sites-available/default:
```diff
@@ -0,0 +1,16 @@
+# Default site configuration
```
Ditto above.

MingcongBai commented on extra-web/nginx/autobuild/postinst:
```diff
@@ -1,2 +1,27 @@
+#!/bin/bash
```
Shebang not necessary.

MingcongBai commented on extra-web/nginx/autobuild/postinst:
```diff
@@ -1,2 +1,27 @@
+
+# Set up permission for logs
 chmod 750 /var/log/nginx
 chown http:log /var/log/nginx
```
Ditto all my notes from httpd postinst above. Don't do out-of-package operations - they don't seem necessary here.

Cyanoxygen committed:
httpd: remove provided config file

Config spliting and sites-enabled inclusion can be done by patching corresponding
file before building.

Cyanoxygen committed:
httpd: remove config file substitution

This change is now done by a patch.

Cyanoxygen committed:
httpd: beyond: replace httpd.conf -> modules.conf

Better explaination with index.html removal

Cyanoxygen committed:
httpd: migrate to sysusers.d

Cyanoxygen committed:
httpd: add /srv/www to tmpfiles.d

Cyanoxygen committed:
httpd: repalce webroot path in default index page

Cyanoxygen committed:
httpd: remove permission handing from postinst

sysusers.d and tmpfiles.d now handles folder and user groups.

Cyanoxygen committed:
mariadb: migrate to sysusers.d

Cyanoxygen committed:
mariadb: add /var/db/mysql to tmpfiles.d

Cyanoxygen committed:
mariadb: simplify default config file

Cyanoxygen committed:
mariadb: modify conditions in postinst

alse improve prompt in postinst

Cyanoxygen committed:
mariadb: add /var/db/mysql within packaging stage

Cyanoxygen committed:
nginx: use patches to provide default configuration file

Cyanoxygen committed:
nginx: add https server example in default virtual host

Cyanoxygen committed:
nginx: let systemd handle webroot directory, user and groups creation

Cyanoxygen committed:
nginx: create directories within packaging stage

Cyanoxygen commented:
Closing, changes are kept to open for another time.
